### PR TITLE
Refactoring Code

### PR DESCRIPTION
**Pull request by RandyHucker**
This refactor mainly deals with moving the Model and Parts data structure returned during a search to just the Fetcher - Before it was duplicated and passed into ThangsAPI into a duplicate structure to complete the import. All the search results have just one instance in memory aside from the UI which needs to use the blender EnumProperty.

Aside from that the repetitive assignments to the 1-8 models were removed and there was a lot of general clean up.

For more improvements:

Re-add back in the folder structure for downloading.
Remove the tuple in the import function and replace with a class and/or the already made Model/Part Struct.
Clean up get_stl_search function for the real time mesh searching on thangs to use the new Model/Parts Struct.

### CODE REVIEW

**Review by garnos — APPROVED** [review]

File: thangs_fetcher.py [hunk 1]
```diff
@@ -666,14 +481,14 @@ def get_stl_search(self, stl_path):
...
-        self.pcoll.ModelView7 = ()
-        self.pcoll.ModelView8 = ()
+        # self.pcoll.ModelView1 = ()
+        # self.pcoll.ModelView2 = ()
+        # self.pcoll.ModelView3 = ()
+        # self.pcoll.ModelView4 = ()
+        # self.pcoll.ModelView5 = ()
+        # self.pcoll.ModelView6 = ()
+        # self.pcoll.ModelView7 = ()
+        # self.pcoll.ModelView8 = ()
```
Comment (BrendenSharp-Physna) [inline thangs_fetcher.py:491]: Small nitpick, delete the commented out code.

File: thangs_fetcher.py [hunk 2]
```diff
@@ -736,12 +554,12 @@ def get_stl_search(self, stl_path):
...
             self.searching = False
             self.newSearch = False
-            self.SelectionFailed = True
+            self.selectionFailed = True
             return
 
         # if os.path.isfile(stl_path):
         #    os.remove(stl_path)
-        #self.Thangs_Utils.clean_downloaded_model_dir("ThangsSelectionSearch")
+        # self.Thangs_Utils.clean_downloaded_model_dir("ThangsSelectionSearch")
```
Comment (BrendenSharp-Physna) [inline thangs_fetcher.py:562]: Delete more commented out code.

File: thangs_fetcher.py [hunk 3]
```diff
@@ -799,7 +617,7 @@ def get_stl_search(self, stl_path):
 
             thumb = self.pcoll.load(modelId, filepath, 'IMAGE')
 
-            self.thumbnailNumbers.append(thumb.icon_id)
+            #self.thumbnailNumbers.append(thumb.icon_id)
```
Comment (BrendenSharp-Physna) [inline thangs_fetcher.py:620]: More commented out code to delete.

**Review by BrendenSharp-Physna — APPROVED** [review]
This is beautiful

File: thangs_fetcher.py (hunk 1 above)
Comment (garnos) [reply thangs_fetcher.py:491]: This is part of the clean up of get_stl_search function that @RandyHucker is doing next.

File: thangs_fetcher.py (hunk 2 above)
Comment (garnos) [reply thangs_fetcher.py:562]: This is part of the clean up of get_stl_search function that @RandyHucker is doing next.

**Review by craigv-physna — APPROVED** [review]
Nice work!  Does it make sense to sit on this until we get the release of the blender addon out, or do we think it's pretty safe to merge?

**Comment by garnos** [comment]
I think it's pretty safe from our testing. Holding off the merge until the release is out too is totally fine.

**Comment by garnos** [comment]
If @ewalker-physna has already gone through testing for development and approved that I say we hold off.

**Pull request merged**